src/server/scripts/Kalimdor/Firelands/firelands.cpp

```diff
@@ -0,0 +1,111 @@
+{
+    public:
+        spell_firelands_smouldering() : SpellScriptLoader("spell_firelands_smouldering") { }
+
+        class spell_firelands_smouldering_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_firelands_smouldering_SpellScript);
+
+            void CheckQuestStatus(std::list<WorldObject*>& targets)
+            {
+                bool raidHasQuest = false;
+
+                targets.remove_if(PlayerCheck());
+                switch (GetSpellInfo()->Id)
+                {
+                    case SPELL_SMOULDERING_QUEST_CHECK_A:
+                        for (auto target : targets)
+                            if (Player* player = target->ToPlayer())
+                                if (player->GetQuestStatus(QUEST_HEART_OF_FLAME_A) == QUEST_STATUS_INCOMPLETE)
+                                    raidHasQuest = true;
+                        break;
+                    case SPELL_SMOULDERING_QUEST_CHECK_H:
+                        for (auto target : targets)
+                            if (Player* player = target->ToPlayer())
+                                if (player->GetQuestStatus(QUEST_HEART_OF_FLAME_H) == QUEST_STATUS_INCOMPLETE)
+                                    raidHasQuest = true;
+                        break;
+                    default:
+                        break;
+                }
+
+                targets.clear();
+                if (raidHasQuest)
+                    targets.push_back(GetCaster());
+            }
+
+            void Register() override
+            {
+                OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_firelands_smouldering_SpellScript::CheckQuestStatus, EFFECT_0, TARGET_UNIT_SRC_AREA_ENTRY);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_firelands_smouldering_SpellScript();
+        }
+};
+
+// http://www.wowhead.com/spell=101093/smouldering
+class spell_firelands_smouldering_aura : public SpellScriptLoader
+{
+    public:
+        spell_firelands_smouldering_aura() : SpellScriptLoader("spell_firelands_smouldering_aura") { }
+
+        class spell_firelands_smouldering_aura_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_firelands_smouldering_aura_SpellScript);
+
+            void SetTarget(WorldObject*& target)
+            {
+                target = GetCaster();
+            }
+
+            void Register() override
+            {
+                OnObjectTargetSelect += SpellObjectTargetSelectFn(spell_firelands_smouldering_aura_SpellScript::SetTarget, EFFECT_0, TARGET_UNIT_NEARBY_ENTRY);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_firelands_smouldering_aura_SpellScript();
+        }
+};
+
+void AddSC_firelands()
+{
+    new spell_firelands_smouldering();
+    new spell_firelands_smouldering_aura();
+}
```

Comment:
I'll fix this and make some other adjustments after I'm done wrapping up one of my other internal commits :)